internal/pkg/service/common/etcdop/watchmirror.go

```diff
@@ -170,11 +183,22 @@ func (s MirrorSetup[T, V]) WithFilter(fn func(event WatchEventT[T]) bool) Mirror
 	return s
 }
 
-func (s MirrorSetup[T, V]) WithOnUpdate(fn func(changes MirrorUpdatedKeys[V])) MirrorSetup[T, V] {
+// WithOnUpdate callback is triggered on each atomic tree update.
+// The update argument contains actual etcd revision,
+// on which the mirror is synchronized.
+func (s MirrorSetup[T, V]) WithOnUpdate(fn func(update MirrorUpdate)) MirrorSetup[T, V] {
 	s.onUpdate = append(s.onUpdate, fn)
 	return s
 }
```

Comment:
Added `WithOnUpdate` method/callback, ... for use-cases when the list of changes is not needed.